src/schema_registry/user.py

```diff
@@ -0,0 +1,36 @@
+"""
+Copyright (c) 2024 Aiven Ltd
+See LICENSE for details
+"""
+
+from dependency_injector.wiring import inject, Provide
+from fastapi import Depends, HTTPException, status
+from fastapi.security import HTTPBasic, HTTPBasicCredentials
+from karapace.auth import AuthenticationError, AuthenticatorAndAuthorizer, User
+from schema_registry.container import SchemaRegistryContainer
+from typing import Annotated
+
+
+@inject
+async def get_current_user(
+    credentials: Annotated[HTTPBasicCredentials, Depends(HTTPBasic(auto_error=False))],
+    authorizer: AuthenticatorAndAuthorizer = Depends(Provide[SchemaRegistryContainer.karapace_container.authorizer]),
+) -> User:
+    if authorizer and not credentials:
```

Comment:
I am thinking to flip this logic, as now there will always be an authorizer, even if it's the no auth authorizer, so I will switch the operands.